src/runtime_src/core/pcie/linux/shim.cpp

```diff
@@ -1314,84 +1315,36 @@ bool shim::zeroOutDDR()
     return true;
 }
 
-/*
- * xclLoadXclBin()
- */
-int
-shim::
-xclLoadXclBin(const xclBin *buffer)
+int shim::getAxlfObjSize(const axlf *buffer)
```

Comment:
Without looking in much detail, I think most of the functions that take `const axlf* buffer` should take `const xrt::xclbin&` and the xrt::xclbin APIs should be used in calculations.  Please check if the necessary meta data parsing can be done using xrt::xclbin.

Reply:
The changes will be more. We can do that for a separate PR.